indico/modules/events/timetable/legacy.py

```diff
@@ -195,6 +199,24 @@ def serialize_break_entry(self, entry, management=False):
                      'title': break_.title})
         return data
 
+    def _get_vc_joinbuttons(self, entry):
+        event = entry.event
+        vcdata = []
+        for vc_room_assoc in entry.object.vc_room_associations:
+            vc_room = vc_room_assoc.vc_room
+            plugin = vc_room.plugin
+            if plugin:
+                vcdata.append(plugin.render_event_buttons(vc_room, vc_room_assoc))
+        if not vcdata and entry.parent:
+            vcdata = self._get_vc_joinbuttons(entry.parent)
+        if not vcdata and event:
+            for vc_room_assoc in event.vc_room_associations:
+                vc_room = vc_room_assoc.vc_room
+                plugin = vc_room.plugin
+                if plugin:
+                    vcdata.append(plugin.render_event_buttons(vc_room, vc_room_assoc))
```

Comment:
This code looks almost like the code above. It can be refactored to avoid duplicating it...

Reply:
refactored